src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php

```diff
@@ -123,43 +123,61 @@ public function databaseRefreshLando(string $siteName = 'default'): Result
     /**
      * Refresh database on Tugboat.
      */
-    public function databaseRefreshTugboat(): Result
+    public function databaseRefreshTugboat(): ResultData
     {
         $this->io()->title('refresh tugboat databases.');
-        $result = null;
+        $result_data = new ResultData();
+
         foreach (array_keys($this->getAllSitesConfig()) as $siteName) {
             try {
                 $dbPath = $this->databaseDownload($siteName);
-            } catch (TaskException) {
+            } catch (TaskException $te) {
                 $this->yell("$siteName: No database configured. Download/import skipped.");
+                $result_data->append($te->getMessage());
                 // @todo: Should we run a site-install by default?
                 continue;
             }
-            if (!is_string($dbPath) || strlen($dbPath) == 0) {
+            // phpstan doesn't seem to be able to understand that $dbPath will
+            // be undefined if an exception is thrown in try{}, so it complains.
+            // I could not find any identifier that would silence this so I had
+            // to silence the whole line :(.
+            /* @phpstan-ignore-next-line */
+            if (!isset($dbPath) || !is_string($dbPath) || strlen($dbPath) == 0) {
```

Comment:
```suggestion
            if (empty($dbPath) || !is_string($dbPath) || strlen($dbPath) == 0) {
```
How about this?

Reply:
```suggestion
            if (empty($dbPath) || !is_string($dbPath)) {
```

I don't think we need the strlen() anymore.